### ISSUE TITLE
Freemarker native support

### ISSUE THREAD

**Issue by carlosthe19916** [apache/camel-quarkus#223]

**Comment by ppalaga** [comment]
@carlosthe19916 are you working on this one?

**Comment by carlosthe19916** [comment]
@ppalaga I'd like to contribute in some way and I'm really curious about using Quakus and Freemarker in personal projects. I'm not sure if I will achieve the goal of adding a new extension but I'd like to give it a try.

**Comment by ppalaga** [comment]
@carlosthe19916 new extensions are welcome! You may find https://camel.apache.org/camel-quarkus/latest/contributor-guide.html useful.

**Comment by ppalaga** [comment]
@carlosthe19916 are you still interested to look into this?

**Comment by carlosthe19916** [comment]
@ppalaga My apologies for not updating this Issue before. I'm still interested in this for sure. Thb, I've created my very basic version of Native Freemarker extension here https://github.com/project-openubl/xml-builder/blob/master/extensions/freemarker/deployment/src/main/java/org/openublpe/quarkus/freemarker/deployment/FreemarkerProcessor.java. I'm not sure how much different could be using camel

**Comment by carlosthe19916** [comment]
@ppalaga Here I've just created a very basic first draft for the camel-freemarker extension. But I still need to enhance the code https://github.com/carlosthe19916/camel-quarkus/tree/camel-freemarker

**Comment by ppalaga** [comment]
@carlosthe19916 tanks for the effort! If you want an early feedback, please feel free to send a draft pull request https://github.blog/2019-02-14-introducing-draft-pull-requests/

**Comment by ppalaga** [comment]
A JVM-only extension (without a proper test) is underway.

**Comment by vsevel** [comment]
is there any plan to offer freemarker as a quarkus extension (independently from camel), and get the camel-quarkus to depend on it?
this would be useful for people interested in quarkus+freemarker, and not necessarily in camel.

**Comment by ppalaga** [comment]
> is there any plan to offer freemarker as a quarkus extension (independently from camel), and get the camel-quarkus to depend on it?

This would indeed make sense and we'd be eventually ready to depend on such a generic Freemarker extension. I think the Freemarker project would be the best place for this effort. Any help to drive that effort is welcome.

**Comment by vsevel** [comment]
> I think the Freemarker project would be the best place for this effort.

or the quarkus project itself. that is where almost all of the extensions are developed.

**Comment by ppalaga** [comment]
> > I think the Freemarker project would be the best place for this effort.
> 
> or the quarkus project itself. that is where almost all of the extensions are developed.

A Quarkus dependency would be easier maintain for us. Anybody please feel free to propose it in Quarkus or in Freemarker.

**Comment by vsevel** [comment]
> A Quarkus dependency would be easier maintain for us. Anybody please feel free to propose it in Quarkus or in Freemarker.

@ppalaga we have a need for a freemarker extension on our project. if time and energy we may develop the extension, starting from https://github.com/carlosthe19916/camel-quarkus/tree/camel-freemarker, then we will see if the quarkus team is interested. I am not sure yet we have this capacity, but it is definitely a possibility.

**Comment by zhfeng** [comment]
@vsevel so we can leverage the https://github.com/quarkiverse/quarkiverse-freemarker now ?

**Comment by vsevel** [comment]
@zhfeng 0.0.4 should provide a good first version to start playing with. it is the bare minimum. from this original issue, I just did not migrate over the filtering capabilities (include/exclude).
for now I included the doc in the project itself https://github.com/quarkiverse/quarkiverse-freemarker/blob/master/doc/freemarker.adoc, waiting for https://github.com/quarkusio/quarkus/issues/12526
I will have minimal bandwidth to make evolutions, but PR are welcome.
